### PR TITLE
Feature/te 329 project navigator for te

### PR DESCRIPTION
**Pull request by kpanier**
I added a new bundle to use a custom navigator. Currently it was enough to use only framework stuff to get the same results as with the project navigator. The Java related error markers are disabled, by excluding Java content.
After this pull request is accepted, we can make a new to disable the Resource Navigator. To Do this we have to change the application.xmi. This should be communicated to avoid merge problems.

### CODE REVIEW

File: rcp/org.testeditor.rcp4.views.projectexplorer/build.properties [hunk 1]
```diff
@@ -0,0 +1,7 @@
+output.. = bin/
+bin.includes = META-INF/,\
+               .,\
+               plugin.xml,\
+               fragment.e4xmi,\
+               icons/
+source.. = src/
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.views.projectexplorer/build.properties]: `xtend-gen` is missing here

**Comment by kpanier** [comment]
xtend-gen added to the build

File: rcp/org.testeditor.rcp4.views.projectexplorer/src/org/testeditor/rcp4/views/projectexplorer/NavFilter.xtend [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import org.eclipse.core.resources.IFile
+import org.eclipse.core.resources.IFolder
+import org.eclipse.jdt.core.ICompilationUnit
+import org.eclipse.jdt.internal.core.JarPackageFragmentRoot
+import org.eclipse.jdt.internal.core.JavaElement
+import org.eclipse.jface.viewers.Viewer
+import org.eclipse.jface.viewers.ViewerFilter
+
+/** filter all classpath containers (Java RT, JUnit) and java elements from navigator
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.views.projectexplorer/src/org/testeditor/rcp4/views/projectexplorer/NavFilter.xtend]: Comment should begin in the line below and start with an upper-case letter.

File: rcp/org.testeditor.rcp4.views.projectexplorer/src/org/testeditor/rcp4/views/projectexplorer/NavFilter.xtend [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * 
+ *  thus the user of the rcp sees only the files relevant to him in the project explorer
+ */
+class NavFilter extends ViewerFilter {
+	override select(Viewer viewer, Object parentElement, Object element) {
+		if (element instanceof ICompilationUnit) { // hide all java compilation units
+			return false
+		}
+		if (element instanceof IFolder) { // don't show any folders that hold generated artifacts (-gen) or maven artifacts (target)
+			if (element.projectRelativePath.segments.exists[equals("bin")||equals("target") || endsWith("-gen")]) {
```
Comment (franzbecker) [inline rcp/org.testeditor.rcp4.views.projectexplorer/src/org/testeditor/rcp4/views/projectexplorer/NavFilter.xtend]: Do we need to check all `segements` and not just `lastSegment`? I think if the parent element is filtered out, the children won't be visible as well.

Comment (kpanier) [inline rcp/org.testeditor.rcp4.views.projectexplorer/src/org/testeditor/rcp4/views/projectexplorer/NavFilter.xtend]: We filter first the parent as the last segment. So we can't get the children.

Comment (franzbecker) [inline rcp/org.testeditor.rcp4.views.projectexplorer/src/org/testeditor/rcp4/views/projectexplorer/NavFilter.xtend]: So it should be sufficient to check `element.projectRelativePath.lastSegment`, right?

**Pull request merged**